apps/api/src/controllers/admin.controller.ts

```diff
@@ -919,10 +936,31 @@ export function makeAdminController(services: ServiceRegistry) {
             updates,
           );
 
+        let updatedRewards: SelectReward[] = [];
+        if (rewards) {
+          if (updatedCompetition.status === COMPETITION_STATUS.ENDED) {
+            throw new ApiError(
+              400,
+              "Cannot update rewards for an ended competition",
```

Comment:
To simplify the request and the logic for modifying the rewards, they are recreated. If the competition has ended, to avoid mistakenly deleting the winning agents, we simply prevent any modification of the rewards